frontend/src/lifetime/input.ts

```diff
@@ -314,13 +316,13 @@ function keyCodeToKey(code: string): string | null {
 		NumpadAdd: "+",
 		NumpadDecimal: ".",
 	};
-	if (code in mapping) return mapping[code];
+	if (code in MAPPING) return MAPPING[code];
```

Comment:
Do not use in since it checks the prototype chain. Instead
```ts
if (MAPPING[code]) return MAPPING[code];
```
In addition the type of mapping is probably better described as `Record<string, string | undifined>`